Add Maintainers section and responsibilities (#30)
openfga/.github: GOVERNANCE.md

This document outlines a set of policies in order to provide a level playing field and open process for contributors to join the OpenFGA project.

> Governance activities are currently performed manually. In the future we might use [Terraform](https://www.terraform.io/) or similar technology to automate tasks, but we're currently a small team and iterating quickly - for now it makes sense to do things in a way that doesn't scale.

Reviewer: How would we use terraform for this?

Reviewer 2: We might want to use: https://registry.terraform.io/providers/integrations/github/latest/docs to manage our repos, the teams, members and access - that would help make access changes more transparent and auditable, as it would mean for a contributor to be a maintainer, that would be a PR that would be approved by one (or 2 or more) admins.

The @product team is concerned with iterating on the product itself. They govern the product roadmap, product marketing, design principles, user research and subject matter expertise, and other product concerns.

*Responsibilities include:*
- Develop and iterate on the governance model
- Review RFCs or delegate their review to a more relevant team
- Maintain OpenFGA's design principles through the RFC review process
- Clarify OpenFGA's design principles over time
- Collaborate with @core to develop a roadmap that fits OpenFGA's design principles
- Collaborate with @core and @maintainers on user research
- Iterate on the product experience

#### Maintainers

> The @maintainers team consists of users who can contribute _and_ approve PRs. They help to ensure contributions align with the project's goals, facilitate collaboration, and maintain the quality of contributions.

Reviewer 3: Does it make sense to link to https://github.com/openfga/community/blob/main/MAINTAINERS.md instead of creating a Maintainers group?

Reviewer 2: Let's not mention @openfga/core (old team - no longer used).

What we can say is we have several maintainers who collaborate together on different aspects of the project (core server, apis, documentation, sdks, cli & tooling and more) and can who can contribute _and_ approve PRs.

Author: - Replaced @maintainers team with link.
- Removed @core team from the doc.

*Responsibilities include:*
- Contribute based on backlog items
- Review incoming PRs and ensure they meet project standards
- Participate in discussions for features and provide feedback
- Consult with other team members to clarify requirements

> - Collaborate with the @core team on technical feasibility and implementation details

Reviewer 2: ```suggestion
- Collaborate with other maintainers on technical feasibility and implementation details
```

- Provide input to the @product team based on community feedback

> #### Contributors

Reviewer: I think we need to revisit the definition of "Contributor" too.

We seem to be defining the "contributors team", not what a contributor does (e.g. create PRs). We might want to suggest something like:

- For small fixes and typos feel free to create a PR
- Beyond that, please first add an issue describing the problem, explaining you want to work on it and how you want to approach the solution.
- Ideally, wait until a maintainer discusses/approves the path.
- Submit a PR to review

@rhamzeh wdyt?

Reviewer 2: For all this this page I would avoid using teams. I would use maintainers above and link to the Maintainers in community (avoid referencing core or any other team).

For contributors, I would also avoid referencing teams (for now) - we can say, anyone can be a contributor, but for those who do it more often and would like to be part of the project, we can add them with limited access.

Reviewer 2: Maybe we should add an "administrative" team/page which is a list of people that folks can contact (e.g. Andres)

Author: Resumed conversation with @rhamzeh about this doc.
- Removed the "teams" from the doc except for product team.
- Beefed up description to what contributors do instead of "contributors team".
